object.c

```diff
@@ -389,37 +390,65 @@ special_object_p(VALUE obj)
     }
 }
 
-/*
- *  call-seq:
- *     obj.clone(freeze: true) -> an_object
- *
- *  Produces a shallow copy of <i>obj</i>---the instance variables of
- *  <i>obj</i> are copied, but not the objects they reference.
- *  #clone copies the frozen (unless +:freeze+ keyword argument is
- *  given with a false value) state of <i>obj</i>.  See
- *  also the discussion under Object#dup.
- *
- *     class Klass
- *        attr_accessor :str
- *     end
- *     s1 = Klass.new      #=> #<Klass:0x401b3a38>
- *     s1.str = "Hello"    #=> "Hello"
- *     s2 = s1.clone       #=> #<Klass:0x401b3998 @str="Hello">
- *     s2.str[1,4] = "i"   #=> "i"
- *     s1.inspect          #=> "#<Klass:0x401b3a38 @str=\"Hi\">"
- *     s2.inspect          #=> "#<Klass:0x401b3998 @str=\"Hi\">"
- *
- *  This method may have class-specific behavior.  If so, that
- *  behavior will be documented under the #+initialize_copy+ method of
- *  the class.
- */
+static int
+obj_freeze_opt(VALUE freeze)
```

Comment:
Seems this can be shared with `freeze_opt`.

Reply:
build fail by using `freeze_opt` like this 

```c
static VALUE
rb_obj_clone2(rb_execution_context_t *ec, VALUE obj, VALUE freeze)
{
    VALUE argv[1];
    argv[0] = freeze;

    int kwfreeze = freeze_opt(1, argv);

    if (!special_object_p(obj))
        return mutable_obj_clone(obj, kwfreeze);
    return immutable_obj_clone(obj, kwfreeze);
}
```

So, added and used `obj_freeze_opt`.

Reply:
I meant to call `obj_freeze_opt()` from `freeze_opt()`.